pkg/lib/authn/authenticator/password/generator.go

```diff
@@ -170,28 +170,14 @@ func (g *Generator) generateOnce() (string, error) {
 func prepareCharacterSet(policy *config.PasswordPolicyConfig) (string, error) {
 	set := map[characterSet]struct{}{}
 
-	if policy.AlphabetRequired {
-		set[characterSetAlphabet] = struct{}{}
-	}
-	if policy.LowercaseRequired {
-		set[characterSetLowercase] = struct{}{}
-	}
-	if policy.UppercaseRequired {
-		set[characterSetUppercase] = struct{}{}
-	}
-	if policy.DigitRequired {
-		set[characterSetDigit] = struct{}{}
-	}
+	// Default to use alphanumeric as base.
+	set[characterSetAlphabet] = struct{}{}
+	set[characterSetDigit] = struct{}{}
+
 	if policy.SymbolRequired {
 		set[characterSetSymbol] = struct{}{}
 	}
 
-	// Default to alphanumeric if no character set is required.
-	if len(set) == 0 {
-		set[characterSetAlphabet] = struct{}{}
-		set[characterSetDigit] = struct{}{}
-	}
-
```

Comment:
Like in js, line 181 to line 190 are redundant. Remove them.

Reply:
fixed up the change